pandas/core/indexes/base.py

```diff
@@ -490,6 +490,11 @@ def __new__(
         if not copy and isinstance(data, (ABCSeries, Index)):
             refs = data._references
 
+        orig_dtype = None
+        if isinstance(data, (ABCSeries, Index)) and dtype is None:
+            # Avoid losing the dtype when infer string is set
+            orig_dtype = data.dtype
```

Comment:
if we change this (and its an API change), it should be made consistent with pd.array, pd.Series, pd.DataFrame

Reply:
I want to trigger this only for the string option for now, but yes I agree that this should be consistent for Pandas objects